src/lib/libdylink.js

```diff
@@ -648,21 +650,27 @@ var LibraryDylink = {
           {{{ makeSetValue('handle', C_STRUCTS.dso.table_addr, 'tableBase', '*') }}};
           {{{ makeSetValue('handle', C_STRUCTS.dso.table_size, 'metadata.tableSize', 'i32') }}};
         }
+#if PTHREADS
       } else {
+        // Read the values for tableBase and memoryBase from shared memory. The
+        // thread that first loaded the DLL already set these values.
         memoryBase = {{{ makeGetValue('handle', C_STRUCTS.dso.mem_addr, '*') }}};
         tableBase = {{{ makeGetValue('handle', C_STRUCTS.dso.table_addr, '*') }}};
       }
+#endif
 
-      var tableGrowthNeeded = tableBase + metadata.tableSize - {{{ from64Expr('wasmTable.length') }}};
-      if (tableGrowthNeeded > 0) {
+      if (metadata.tableSize) {
+#if ASSERTIONS
+        assert({{{ from64Expr('wasmTable.length') }}} == tableBase, `unexpected table size while loading ${libName}: ${wasmTable.length}`);
+#endif
 #if DYLINK_DEBUG
-        dbg("loadModule: growing table: " + tableGrowthNeeded);
+        dbg("loadModule: growing table by: " + metadata.tableSize);
 #endif
-        wasmTable.grow({{{ toIndexType('tableGrowthNeeded') }}});
+        wasmTable.grow({{{ toIndexType('metadata.tableSize') }}});
```

Comment:
Before we grew  by `tableBase + metadata.tableSize - wasmTable.length`, that is, we grew to size `tablebase + metadata.tableSize`. Now we grow by `metadata.tableSize`, which grows to `wasmTable.length + metadata.tableSize`. Why is that right?

Reply:
There's a new assertion that `tableBase` is equal to `wasmTable.length`.

Reply:
oh, now I see, thanks. makes sense.

Reply:
> So before it would be possible to load `liba.so` and then load `libb.so` on the main thread and load them in the opposite order on the side thread. But this rules that out?

Yes, this was should already have been impossible, and we have various check and assertions that try to verify that this can never happen.

This just adds one more place we that we can assert this invariant.